console/src/channel/CreateCalculated.tsx

```diff
@@ -0,0 +1,288 @@
+// Copyright 2024 Synnax Labs, Inc.
+//
+// Use of this software is governed by the Business Source License included in the file
+// licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with the Business Source
+// License, use of this software will be governed by the Apache License, Version 2.0,
+// included in the file licenses/APL.txt.
+
+import { channel, DataType, Rate } from "@synnaxlabs/client";
+import { MAIN_WINDOW } from "@synnaxlabs/drift";
+import {
+  Align,
+  Button,
+  Channel,
+  Form,
+  Input,
+  Nav,
+  Select,
+  Synnax,
+  Text,
+  Triggers,
+  useSyncedRef,
+} from "@synnaxlabs/pluto";
+import { unique } from "@synnaxlabs/x";
+import { useMutation } from "@tanstack/react-query";
+import * as monaco from "monaco-editor";
+import { type ReactElement, useEffect, useState } from "react";
+import { z } from "zod";
+
+import { Code } from "@/code";
+import { CSS } from "@/css";
+import { Layout } from "@/layout";
+import type { RendererProps } from "@/layout/slice";
+
+export interface CalculatedChannelArgs {
+  channelKey?: number;
+  channelName?: string;
+  expression?: string;
+}
+
+interface CalculatedModalRendererProps extends RendererProps {
+  args?: CalculatedChannelArgs;
+}
+
+export const CREATE_CALCULATED_LAYOUT_TYPE = "createCalculatedChannel";
+
+const SAVE_TRIGGER: Triggers.Trigger = ["Control", "Enter"];
+
+export const createCalculatedLayout: Layout.State = {
+  key: CREATE_CALCULATED_LAYOUT_TYPE,
+  type: CREATE_CALCULATED_LAYOUT_TYPE,
+  windowKey: MAIN_WINDOW,
+  name: "Channel.Create.Calculated",
+  icon: "Channel",
+  location: "mosaic",
+  tab: {
+    closable: true,
+    editable: false,
+  },
+  window: {
+    resizable: false,
+    size: { height: 600, width: 1000 },
+    navTop: true,
+    showTitle: true,
+  },
+};
+
+const schema = channel.newPayload
+  .extend({
+    name: z.string().min(1, "Name must not be empty"),
+    dataType: DataType.z.transform((v) => v.toString()),
+  })
+  .refine((v) => !v.isIndex || new DataType(v.dataType).equals(DataType.TIMESTAMP), {
+    message: "Index channel must have data type TIMESTAMP",
+    path: ["dataType"],
+  })
+  .refine((v) => v.isIndex || v.index !== 0 || v.virtual, {
+    message: "Data channel must have an index",
+    path: ["index"],
+  });
+
+export const CreateCalculatedModal = (({
+  onClose,
+  args,
+}: CalculatedModalRendererProps): ReactElement => {
+
+  const client = Synnax.use();
+
+  const [initialLoading, setInitialLoading] = useState(!!args?.channelKey);
+
+  const methods = Form.use<typeof schema>({
+    schema,
+    values: {
+      key: args?.channelKey ?? 0,
+      name: args?.channelName ?? "",
+      index: 0,
+      dataType: "float32",
+      internal: false,
+      isIndex: false,
+      leaseholder: 0,
+      rate: Rate.hz(0),
+      virtual: true,
+      expression: args?.expression ?? "result = np.array([])",
+      requires: [],
+    },
+  });
+
+  useEffect(() => {
```

Comment:
There's no need to use an effect here. Much better to use tanstacks `useQuery` and create a wrapper that loads in the initial form values. Let's walk through this verbally.